lib/src/main/java/org/team1540/base/preferencemanager/PreferenceManager.java

```diff
@@ -115,14 +119,17 @@ private synchronized void run() {
     if (enabled) {
       for (PreferenceField preference : preferences) {
         try {
-          if (!Preferences.getInstance().containsKey(preference.label)) {
+          if ((preference.persistent && !Preferences.getInstance().containsKey(preference.label)) || (!preference.persistent && !SmartDashboard.containsKey(preference.label))) {
+            BiConsumer putFunc = preference.persistent ? preference.type.putFunction
+                : preference.type.nonPersistentPutFunction;
             //noinspection unchecked
-            preference.type.putFunction
-                .accept(preference.label, preference.field.get(preference.obj));
+            putFunc.accept(preference.label, preference.field.get(preference.obj));
           } else {
+            BiFunction getFunc = preference.persistent ? preference.type.getFunction
+                : preference.type.nonPersistentGetFunction;
             //noinspection unchecked
-            preference.field.set(preference.obj, preference.type.getFunction
-                .apply(preference.label, preference.field.get(preference.obj)));
+            preference.field.set(preference.obj,
+                getFunc.apply(preference.label, preference.field.get(preference.obj)));
```

Comment:
Why not just have the ternary inline here?

Reply:
Mostly to keep the line length to merely ridiculous levels instead of unconscionable ones.

Reply:
Bad reason, decreases readability. Lines should have a set width anyways.